controllers/machinedeployment_sync.go

```diff
@@ -110,6 +110,7 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(d *clusterv1.MachineDeplo
 
 		// Set existing new machine set's annotation
 		annotationsUpdated := mdutil.SetNewMachineSetAnnotations(d, msCopy, newRevision, true, logger)
+		clusterv1.PopulateDefaultsMachineSet(msCopy)
```

Comment:
Shouldn't this be taken care from the defaulting webhook?

Reply:
Yup. fixed now